ops-catalog/generate-migration.sh

```diff
@@ -19,36 +19,118 @@ if [[ -z "$ENVIRONMENT" ]]; then
 # For postgres, this is done by doubling the single quote character (replace ' with '').
-BUNDLED_CATALOG="$(flowctl raw bundle --source "$INPUT_FILENAME" | sed "s/'/''/g")"
+BUNDLED_TENANT="$(flowctl raw bundle --source "${SCRIPT_DIR}/template-tenant.flow.yaml" | sed "s/'/''/g")"
+BUNDLED_LEVEL_1_DERIVATION="$(flowctl raw bundle --source "${SCRIPT_DIR}/template-level-1-derivation.flow.yaml" | sed "s/'/''/g")"
+BUNDLED_LEVEL_2_DERIVATION="$(flowctl raw bundle --source "${SCRIPT_DIR}/template-level-2-derivation.flow.yaml" | sed "s/'/''/g")"
+BUNDLED_MATERIALIZATION="$(flowctl raw bundle --source "$MATERIALIZATION_TEMPLATE_FILENAME" | sed "s/'/''/g")"
+
 
 cat << EOF
 -- This migration was generated by ops-catalog/generate-migration.sh
--- It updates the ops catalog template that's used by agent when provisioning new tenants,
--- and also re-publishes the ops catalogs for all existing tenants.
+-- It updates the ops catalog templates that are used by agent when
+-- provisioning new tenants, and also re-publishes the ops catalogs
+-- for all existing tenants.
 begin;
 
 do \$\$
 declare
-	bundled_catalog_arg json := '${BUNDLED_CATALOG}';
+	new_draft_id flowid := internal.id_generator();
+	publication_id flowid := internal.id_generator();
 	ops_user_id uuid;
 	current_tenant tenants;
+	tenant_count integer;
+	current_l1_stat_rollup integer;
+	tenant_spec jsonb;
+	l1_derivation_specs jsonb = '{}';
+	l2_derivation_spec jsonb;
+	materialization_spec jsonb;
 begin
 
 	-- Identify user which owns ops specifications.
 	select id into strict ops_user_id from auth.users where email = 'support@estuary.dev';
 
-	-- Update the ops catalog template.
-	update ops_catalog_template set bundled_catalog = bundled_catalog_arg;
+	-- Update the ops catalog templates.
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_TENANT}' where template_type = 'tenant';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_LEVEL_1_DERIVATION}' where template_type = 'l1_derivation';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_LEVEL_2_DERIVATION}' where template_type = 'l2_derivation';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_MATERIALIZATION}' where template_type = 'materialization';
+
+	-- There must be at least one tenant in the system to proceed. This is a guard for local
+	-- development when first starting the local stack.
+	select count(*)
+	into strict tenant_count
+	from tenants;
+	if tenant_count = 0 then
+		return;
+	end if;
+
+	-- Create a draft of ops changes.
+	insert into drafts (id, user_id, detail) values
+	(new_draft_id, ops_user_id, 're-publishing ops catalog');
+
+	-- Queue a publication of the draft.
+	insert into publications (id, user_id, draft_id) values
+	(publication_id, ops_user_id, new_draft_id);
 
-	-- Start a publication of the updated template for each existing tenant.
+	-- Handle the tenant-specific specs for each tenant.
```

Comment:
This is all going into a single draft. There will be specs for every single tenant. I'm not sure if there's a limit to how many specs can be in a single draft. It was a little easier to do it this way, but if we are worried about a single draft becoming too large I could create a separate draft for each tenant's specific specs.